imod/protocols/protocol_goldBeadPicker3d.py

```diff
@@ -155,45 +154,49 @@ def pickGoldBeadsStep(self, tsId):
             self.runProgram('model2point', paramsModel2Point)
 
         except Exception as e:
-            self._failedTomos.append(tsId)
+            self._failedItems.append(tsId)
             self.error(f"findbeads3d or model2point execution failed for tsId {tsId} -> {e}")
 
     def createOutputStep(self, tsId):
         tomo = self.tomoDict[tsId]
-        if tsId in self._failedTomos:
-            self.createOutputFailedSet(tomo)
-        else:
-            coordFilePath = self.getExtraOutFile(tsId, ext=XYZ_EXT)
-            if os.path.exists(coordFilePath):
-                beadDiam = self.beadDiameter.get()
-                coordList = utils.formatGoldBead3DCoordinatesList(coordFilePath)
-                output = self.getOutputSetOfCoordinates3Ds(self.getInputSet(),
-                                                           self.getInputSet())
-
-                for element in coordList:
-                    newCoord3D = Coordinate3D()
-                    newCoord3D.setVolume(tomo)
-                    newCoord3D.setX(element[0], constants.BOTTOM_LEFT_CORNER)
-                    newCoord3D.setY(element[1], constants.BOTTOM_LEFT_CORNER)
-                    newCoord3D.setZ(element[2], constants.BOTTOM_LEFT_CORNER)
-
-                    # newCoord3D.setVolId(tsObjId)
-                    output.append(newCoord3D)
-                    output.update(newCoord3D)
+        with self._lock:
+            if tsId in self._failedItems:
+                self.createOutputFailedSet(tomo)
+            else:
+                coordFilePath = self.getExtraOutFile(tsId, ext=XYZ_EXT)
+                if os.path.exists(coordFilePath):
+                    beadDiam = self.beadDiameter.get()
+                    coordList = utils.formatGoldBead3DCoordinatesList(coordFilePath)
+                    output = self.getOutputSetOfCoordinates3Ds(self.getInputSet(pointer=True),
+                                                               self.getInputSet())
                     output.setBoxSize(beadDiam)
-                    output.write()
 
-                self._store(output)
+                    for element in coordList:
+                        newCoord3D = Coordinate3D()
+                        newCoord3D.setVolume(tomo)
+                        newCoord3D.setX(element[0], constants.BOTTOM_LEFT_CORNER)
+                        newCoord3D.setY(element[1], constants.BOTTOM_LEFT_CORNER)
+                        newCoord3D.setZ(element[2], constants.BOTTOM_LEFT_CORNER)
+
+                        output.append(newCoord3D)
+                        output.update(newCoord3D)
+                    output.write()
+                    self._store(output)
```

Comment:
I can see you put back these two:
output.write()
self._store(output)
Is there a reason? Protocol works without these redundant calls. Write and store are called in the closeOutputSet step for all outputs